core/Pos/Data/Attributes.hs

```diff
@@ -1,68 +1,93 @@
+{-# LANGUAGE ScopedTypeVariables #-}
 -- | Helper data type for block, tx attributes.
 --
 -- Map with integer 1-byte keys, arbitrary-type polymorph values.
 -- Needed primarily for partial serialization. Values are either
 -- parsed and put to some constructor or left as unparsed.
 
 module Pos.Data.Attributes
-       ( Attributes (..)
+       ( UnparsedFields(..)
+       , fromRaw
+       , toRaw
+       , Attributes (..)
        , areAttributesKnown
-       , getAttributes
-       , putAttributes
-       , putAttributesS
-       , sizeAttributes
+       , encodeAttributes
+       , decodeAttributes
        , mkAttributes
        ) where
 
 import           Universum
 
 import qualified Data.ByteString     as BS
 import           Data.Default        (Default (..))
+import qualified Data.Hashable       as H
+import qualified Data.Map            as M
 import           Data.Text.Buildable (Buildable)
 import qualified Data.Text.Buildable as Buildable
 import           Formatting          (bprint, build, int, (%))
 import qualified Prelude
 
-import           Pos.Binary.Class    (Peek, Poke, PokeWithSize (..), getBytes,
-                                      getPeekLength, getWithLength, getWithLengthLimited,
-                                      getWord8, lookAhead, putBytesS, putWithLengthS,
-                                      putWord8S)
+import           Pos.Binary.Class
+
+-- | Representation of unparsed fields in Attributes. Newtype wrapper is used
+-- for clear backward compatibility between previous representation (which was
+-- just a single ByteString) during transition from Store to CBOR.
+newtype UnparsedFields = UnparsedFields (Map Word8 ByteString)
+    deriving (Eq, Ord, Show, Generic, Typeable, NFData)
+
+instance Hashable UnparsedFields where
+    hashWithSalt salt = H.hashWithSalt salt . M.toList . fromUnparsedFields
+
+fromUnparsedFields :: UnparsedFields -> Map Word8 ByteString
+fromUnparsedFields (UnparsedFields m) = m
+
+-- | Convert from Store format.
```

Comment:
Also I don't understand why these functions are needed and they are not used (I checked using `scripts/grep.sh`), so I think they should be removed with these comments.